### PR TITLE
[v2.7] K8s june 2023 patches

### PR DESCRIPTION
**Pull request by vardhaman22**
linked issue: https://github.com/rancher/rancher/issues/41751

### CODE REVIEW

File: pkg/rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -8770,6 +8770,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/mirrored-pause:3.7",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.21.1",
 		},
+		// Enabled out of band post v2.7.5
+		"v1.24.15-rancher1-1": {
+			Etcd:                      "rancher/mirrored-coreos-etcd:v3.5.6",
```
Comment (kinarashah) [inline pkg/rke/k8s_rke_system_images.go]: Is this etcd bump intentional? Technically there shouldn't be issues but it'd be safer to keep patch versions at the same etcd level unless there's a specific issue we're trying to fix.

**Review by kinarashah — DISMISSED** [review]
Other changes LGTM

File: pkg/rke/k8s_rke_system_images.go (hunk 1 above)
Comment (doflamingo721) [reply pkg/rke/k8s_rke_system_images.go]: Downgraded the etcd version to 3.5.4

**Review by kinarashah — APPROVED** [review]
Thanks @doflamingo721 for also syncing in the v2.7.5 release changes. LGTM.

**Pull request merged**